### ISSUE TITLE
Support Git short hashes for GitHub URLs.

### ISSUE THREAD

**Issue by ngokevin** [jsdelivr/jsdelivr#18100]
Right now, JSDelivr only supports full hash for GitHub URLs (`username/repo@FULL_HASH`). Would be nice to support the short hash (7 chars, `git rev-parse --short HEAD`) (as RawGit did).

**Comment by MartinKolarik** [comment]
I agree the URLs would look nicer but it would introduce a lot of ambiguity which we'd like to avoid. Right now we know that any 40 hex character long version should be treated as a commit hash. If we allowed shorter hashes, we'd have to get a list of releases/tags from GitHub and check if the short hash isn't actually a release/tag, then check if the short hash maps to an existing full hash and only then request the file.

Also, this problem shouldn't be very common as you can use regular releases with most repos.
